mws/acceptance/cmk_test.go

```diff
@@ -0,0 +1,105 @@
+package acceptance
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/databrickslabs/databricks-terraform/common"
+	"github.com/databrickslabs/databricks-terraform/internal/acceptance"
+	"github.com/databrickslabs/databricks-terraform/internal/qa"
+	. "github.com/databrickslabs/databricks-terraform/mws"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestMwsAccCustomerManagedKeys(t *testing.T) {
+	cloudEnv := os.Getenv("CLOUD_ENV")
+	if cloudEnv != "MWS" {
+		t.Skip("Acceptance tests skipped unless CLOUD_ENV=MWS is set")
+	}
+	cmkConfig := qa.EnvironmentTemplate(t, `
+	provider "databricks" {
+		host     = "{env.DATABRICKS_HOST}"
+		username = "{env.DATABRICKS_USERNAME}"
+		password = "{env.DATABRICKS_PASSWORD}"
+	}
+	resource "databricks_mws_customer_managed_keys" "my_cmk" {
+		account_id   = "{env.DATABRICKS_ACCOUNT_ID}"
+		aws_key_info {
+			key_arn   = "{env.TEST_KMS_KEY_ARN}"
+			key_alias = "{env.TEST_KMS_KEY_ALIAS}"
+		}
+	}`)
+	var customerManagedKey CustomerManagedKey
+	acceptance.AccTest(t, resource.TestCase{
+		CheckDestroy: testMWSNetworkResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: cmkConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testMWSCustomerManagedKeyResourceExists("databricks_mws_customer_managed_keys.my_cmk", &customerManagedKey, t),
+				),
+				Destroy: false,
+			},
+			{
+				PreConfig: func() {
+					conn := common.CommonEnvironmentClient()
+					err := NewCustomerManagedKeysAPI(conn).Delete(customerManagedKey.AccountID,
+						customerManagedKey.CustomerManagedKeyID)
+					if err != nil {
+						panic(err)
+					}
+				},
+				Config: cmkConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testMWSCustomerManagedKeyResourceExists("databricks_mws_customer_managed_keys.my_cmk", &customerManagedKey, t),
+				),
+				ExpectNonEmptyPlan: false,
+				Destroy:            false,
+			},
+			{
+				PreConfig: func() {
+					conn := common.CommonEnvironmentClient()
+					err := NewCustomerManagedKeysAPI(conn).Delete(customerManagedKey.AccountID,
+						customerManagedKey.CustomerManagedKeyID)
+					if err != nil {
+						panic(err)
+					}
+				},
+				Config: cmkConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testMWSCustomerManagedKeyResourceExists("databricks_mws_customer_managed_keys.my_cmk", &customerManagedKey, t),
+				),
+				ExpectNonEmptyPlan: false,
+				Destroy:            false,
```

Comment:
why do we have to leave resources dangling?

Reply:
fixed